Index: src/net/wimpi/pim/contact/model/GeographicalInformation.java
===================================================================
RCS file: /cvsroot/jpim/jpim/src/net/wimpi/pim/contact/model/GeographicalInformation.java,v
retrieving revision 1.2
diff -u -r1.2 GeographicalInformation.java
--- src/net/wimpi/pim/contact/model/GeographicalInformation.java	25 Jun 2003 19:53:46 -0000	1.2
+++ src/net/wimpi/pim/contact/model/GeographicalInformation.java	10 May 2008 21:28:30 -0000
@@ -9,6 +9,7 @@
 package net.wimpi.pim.contact.model;
 
 import java.io.Serializable;
+import java.math.BigDecimal;
 import java.util.TimeZone;
 
 /**
@@ -52,7 +53,7 @@
    *
    * @return the longitude as <tt>double</tt>.
    */
-  public double getLongitude();
+  public BigDecimal getLongitude();
 
   /**
    * Sets the longitude of the geographical
@@ -63,7 +64,7 @@
    *
    * @param longitude the longitude as <tt>double</tt>.
    */
-  public void setLongitude(double longitude);
+  public void setLongitude(BigDecimal longitude);
 
   /**
    * Returns the latitude of the geographical
@@ -74,7 +75,7 @@
    *
    * @return the latitude as <tt>double</tt>.
    */
-  public double getLatitude();
+  public BigDecimal getLatitude();
 
   /**
    * Sets the latitude of the geographical
@@ -85,6 +86,6 @@
    *
    * @param latitude the latitude as <tt>double</tt>.
    */
-  public void setLatitude(double latitude);
+  public void setLatitude(BigDecimal latitude);
 
 }//interface GeographicalInformation
Index: src/net/wimpi/pim/util/versitio/versitInputStream.java
===================================================================
RCS file: /cvsroot/jpim/jpim/src/net/wimpi/pim/util/versitio/versitInputStream.java,v
retrieving revision 1.5
diff -u -r1.5 versitInputStream.java
--- src/net/wimpi/pim/util/versitio/versitInputStream.java	16 Jul 2003 17:32:04 -0000	1.5
+++ src/net/wimpi/pim/util/versitio/versitInputStream.java	10 May 2008 21:28:30 -0000
@@ -20,6 +20,7 @@
   private BufferedReader m_Input;
   private boolean m_hasLine = true;
   private boolean m_Unread = false;
+  private boolean m_normalFoldingDetected = false;
   private int m_Char;
 
   public versitInputStream(InputStream in) {
@@ -45,7 +46,11 @@
         if (b == EQUAL) {
           //lookahead character
           b = readCharacter();
-          if (b == CR || b == LF) {
+          if ((b == CR || b == LF) && !m_normalFoldingDetected) {
+            // this means that we have the so-called 'evolution-style' folding
+            // and we haven't encountered any lines that would begin with a 
+            // whitespace, which would indicate that this file followis
+            // 'normal' folding rules
             trail = true;
           } else {
             trail = false;
@@ -67,6 +72,12 @@
             b = readCharacter();
             if (b == SPACE || b == HTAB) {
               //standard "line folder" swallow it
+              //note down the fact that standard folding has been encountered
+              //from now-one '=' signs at the end of the line will be treated
+              //as normal signs and not as folding indicators
+              //this should do the trick for Base64 encoded attachments with
+              //the equal sign as padding
+              m_normalFoldingDetected = true;
               continue;
             } else if (b == CR || b == LF) {
               throw new IOException("Linebreaks mangled.");
Index: src/net/wimpi/pim/contact/io/vcard/GEOItemHandler.java
===================================================================
RCS file: /cvsroot/jpim/jpim/src/net/wimpi/pim/contact/io/vcard/GEOItemHandler.java,v
retrieving revision 1.3
diff -u -r1.3 GEOItemHandler.java
--- src/net/wimpi/pim/contact/io/vcard/GEOItemHandler.java	15 Jul 2003 21:32:36 -0000	1.3
+++ src/net/wimpi/pim/contact/io/vcard/GEOItemHandler.java	10 May 2008 21:28:30 -0000
@@ -8,6 +8,8 @@
  ***/
 package net.wimpi.pim.contact.io.vcard;
 
+import java.math.BigDecimal;
+
 import net.wimpi.pim.Pim;
 import net.wimpi.pim.contact.model.Contact;
 import net.wimpi.pim.contact.model.GeographicalInformation;
@@ -38,9 +40,9 @@
       geo = Pim.getContactModelFactory().createGeographicalInformation();
       ct.setGeographicalInformation(geo);
     }
-    String[] comps = StringUtil.split(item.getDecodedValue(), ",");
-    geo.setLatitude(Double.parseDouble(comps[0]));
-    geo.setLongitude(Double.parseDouble(comps[1]));
+    String[] comps = StringUtil.split(item.getDecodedValue(), ";");
+    geo.setLatitude(new BigDecimal(comps[0]));
+    geo.setLongitude(new BigDecimal(comps[1]));
   }//handleItem
 
   public versitItem[] createItems(Contact ct) {
@@ -49,9 +51,9 @@
       return new versitItem[0];
     }
     versitItem[] items = new versitItem[1];
-    double longitude = geo.getLongitude();
-    double latitude = geo.getLatitude();
-    if (longitude == 0 && latitude == 0) {
+    BigDecimal longitude = geo.getLongitude();
+    BigDecimal latitude = geo.getLatitude();
+    if (longitude.compareTo(BigDecimal.ZERO) == 0 && latitude.compareTo(BigDecimal.ZERO) == 0) {
       return new versitItem[0];
     }
     versitItem item = new versitItem(versitToken.GEO);
Index: src/net/wimpi/pim/contact/basicimpl/GeographicalInformationImpl.java
===================================================================
RCS file: /cvsroot/jpim/jpim/src/net/wimpi/pim/contact/basicimpl/GeographicalInformationImpl.java,v
retrieving revision 1.2
diff -u -r1.2 GeographicalInformationImpl.java
--- src/net/wimpi/pim/contact/basicimpl/GeographicalInformationImpl.java	15 Jul 2003 23:53:49 -0000	1.2
+++ src/net/wimpi/pim/contact/basicimpl/GeographicalInformationImpl.java	10 May 2008 21:28:30 -0000
@@ -8,6 +8,7 @@
  ***/
 package net.wimpi.pim.contact.basicimpl;
 
+import java.math.BigDecimal;
 import java.util.TimeZone;
 
 import net.wimpi.pim.contact.model.GeographicalInformation;
@@ -27,8 +28,8 @@
 
   //instance attributes
   protected TimeZone m_TZ = TimeZone.getDefault();
-  protected double m_Longitude;
-  protected double m_Latitude;
+  protected BigDecimal m_Longitude;
+  protected BigDecimal m_Latitude;
 
   public TimeZone getTimeZone() {
     return m_TZ;
@@ -38,19 +39,19 @@
     m_TZ = timezone;
   }//setTimeZone
 
-  public double getLongitude() {
+  public BigDecimal getLongitude() {
     return m_Longitude;
   }//getLongitude
 
-  public void setLongitude(double longitude) {
+  public void setLongitude(BigDecimal longitude) {
     m_Longitude = longitude;
   }//setLongitude
 
-  public double getLatitude() {
+  public BigDecimal getLatitude() {
     return m_Latitude;
   }//getLatitude
 
-  public void setLatitude(double latitude) {
+  public void setLatitude(BigDecimal latitude) {
     m_Latitude = latitude;
   }//setLatitude
 
Index: .project
===================================================================
RCS file: .project
diff -N .project
--- /dev/null	1 Jan 1970 00:00:00 -0000
+++ .project	1 Jan 1970 00:00:00 -0000
@@ -0,0 +1,17 @@
+<?xml version="1.0" encoding="UTF-8"?>
+<projectDescription>
+	<name>jpim-0.1</name>
+	<comment></comment>
+	<projects>
+	</projects>
+	<buildSpec>
+		<buildCommand>
+			<name>org.eclipse.jdt.core.javabuilder</name>
+			<arguments>
+			</arguments>
+		</buildCommand>
+	</buildSpec>
+	<natures>
+		<nature>org.eclipse.jdt.core.javanature</nature>
+	</natures>
+</projectDescription>
Index: .classpath
===================================================================
RCS file: .classpath
diff -N .classpath
--- /dev/null	1 Jan 1970 00:00:00 -0000
+++ .classpath	1 Jan 1970 00:00:00 -0000
@@ -0,0 +1,6 @@
+<?xml version="1.0" encoding="UTF-8"?>
+<classpath>
+	<classpathentry kind="src" path="src"/>
+	<classpathentry kind="con" path="org.eclipse.jdt.launching.JRE_CONTAINER"/>
+	<classpathentry kind="output" path="build_eclipse"/>
+</classpath>
